WSTEAMA-376 - Adding component healthcheck to Most Read component (#10898)
bbc/simorgh: src/app/components/MostRead/README.md

Reviewer: Can we remove the #Most Read title from the readme, as it is causing the title to appear twice in Storybook (it's built into Component Health)

## Description

> This component renders a collection of most read articles to a news page so that relevant and trusted onward journeys are provided for the audience. It is featured on Homepages, Topic, Article, and Story pages and comprises a number rank, a title, and a link to the article. These elements sit within a region landmark. The component uses `visualStyle` and `visualProminence` to determine the number of ranked items and its positioning on the page. e.g `visualStyle` = `RANKED` & `visualStyle` = `LOW` will display 5 ranked items. While `visualStyle` of `RANKED` and `visualStyle` of `NORMAL` will display 10 ranked items.

Reviewer: This is not quite true - the number of items to display is actually based on the service configuration. Although that might actually be a decent enhancement - do you perhaps want to follow up with Product to see if that's something they wish to implement in future?

Also feel free to re-word this 😄 

```suggestion
This component renders a collection of most read articles to a news page so that relevant and trusted onward journeys are provided for the audience. It is featured on Homepages, Topic, Article, and Story pages and comprises a number rank, a title, and a link to the article. These elements sit within a region landmark. The component limits the number of Most Read items to be displayed based on the [`mostRead -> numberOfItems`](https://github.com/bbc/simorgh/blob/561414ae7c1e6636372381a8e0deddf48f926c1c/src/app/lib/config/services/mundo.ts#L256-L257) value for each service.
```

Author: Ah, this is what's written on the main story ticket, I wonder if either that's wrong or something was changed along the way. I'll follow up and find out!

Reviewer: Ah OK. It's definitely not how the component was originally implemented, but I think it's a good idea - it gives control to editorial without having to make a code change.

However I suggest we leave as is for the moment, and if this is desired behaviour, let's get a ticket created, so that we can do it later.

> The component uses a [MostReadData](https://github.com/bbc/simorgh/blob/c4d38ae16587f29306d1109ea19bd55205c5709b/src/app/components/MostRead/types.ts#L79-L90) type, containing a simplified version of data which is passed in and used to render the information on the page. This data contains an items array with the following fields: id, rank, title, href, and timestamp.

Reviewer: I guess we don't really need to mention about the data being a simplified version - it will soon become the only version of the data :) 
```suggestion
The component uses a [MostReadData](https://github.com/bbc/simorgh/blob/c4d38ae16587f29306d1109ea19bd55205c5709b/src/app/components/MostRead/types.ts#L79-L90) type, which contains an items array with the following fields: id, rank, title, href, and timestamp.
```

> It also uses a [ColumnLayout](https://github.com/bbc/simorgh/blob/c4d38ae16587f29306d1109ea19bd55205c5709b/src/app/components/MostRead/types.ts#L7) type that determines how many columns the component is spread across: `oneColumn`, `twoColumn` or `multiColumn`.

Reviewer: Is it worth mentioning that `multiColumn` spreads the data across 5 columns?

## Props

| Name         | type         | Description                                                                           |
| ------------ | ------------ | ------------------------------------------------------------------------------------- |
| data         | MostReadData | Contains the data rendered to the browser. i.e rank, title, and appropriate href link |

> | columnLayout | ColumnLayout | Determines how many columns the component is spread across. Defaults to `twoColumn`.  |

Reviewer: The default is `multiColumn` -> https://github.com/bbc/simorgh/blob/561414ae7c1e6636372381a8e0deddf48f926c1c/src/app/components/MostRead/index.tsx#L41-L42

```suggestion
| columnLayout | ColumnLayout | Determines how many columns the component is spread across. Defaults to `twoColumn`.  |
```

Reviewer: Ooh this reminds me that we also need to add `size` as a parameter, including description, values & default.

## Description

> This component renders a collection of most read articles to a news page so that relevant and trusted onward journeys are provided for the audience. It is featured on Homepages, Topic, Article, and Story pages and comprises a number rank, a title, and a link to the article. These elements sit within a region landmark. The component uses `visualStyle` and `visualProminence` to determine the number of ranked items and its positioning on the page. e.g `visualStyle` = `RANKED` & `visualStyle` = `LOW` will display 5 ranked items. While `visualStyle` of `RANKED` and `visualStyle` of `NORMAL` will display 10 ranked items.

Reviewer: I am not seeing this in Storybook - let's see if there's something missing from the stories file

Author: I think I can see where I've missed something. Will resolve now!

Author: Sorted and appearing in storybook 🎉
